### ISSUE TITLE
Feature: Add visual indicator when Mouse control is toggled On

### ISSUE THREAD

**Issue by samrocketman** [endless-sky/endless-sky#8339]
### Problem Description

While controlling a ship it is a bit difficult to tell sometimes whether or not mouse control is enabled.  Sometimes you have to wiggle the mouse to confirm that it is on or off.  It would be nice to have a visual indicator of speed up.

### Related Issue Links

This feature request adds on to this now-merged pull request

* https://github.com/endless-sky/endless-sky/pull/6936

### Desired Solution

A small icon to indicate mouse control is enabled.  The following is a UI mock of what I want similar to "speed up time" toggle.

![Screenshot taken at 2023-02-10_13-09-38](https://user-images.githubusercontent.com/875669/218165603-71385808-e10a-49c5-a872-01693d250199.png)

### Alternative Approaches

None considered.

### Additional Context

I've found this to be a struggle while playing on alternative platforms such as Steam Deck

**Comment by tibetiroka** [comment]
I think using an actual cursor icon might be confusing for people who frequently lose their cursor. Also, I'm not sure how much clearance we have at the alert indicator.

I love the idea though.

**Comment by samrocketman** [comment]
That's fair open to suggested alternatives.  This is why I opened the issue instead of coding right away (so that I don't create something undesirable from the get go).

If you can think of another alternative that works I am open to it.  I naturally look to the small icon for "fast forward" and would do the same for a mouse icon.

In this case, the mouse icon is static.  It does not move and remains in the same spot like the fast forward icon.

**Comment by Zireael07** [comment]
I'd use a mouse icon such as https://www.flaticon.com/free-icon/mouse-clicker_9635625?term=mouse&page=1&position=3&origin=tag&related_id=9635625 instead of something that can be mistaken for cursor

**Comment by tibetiroka** [comment]
Maybe something like [this](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse1.mm.bing.net%2Fth%3Fid%3DOIP.I-D8RmS4-TPFGRo1wKtAzwHaHa%26pid%3DApi&f=1&ipt=13ef12944fde078908b70d81a9a1f000afc0b0991043012dba7d340d0c5ffef9&ipo=images) could work as well.

**Comment by samrocketman** [comment]
Either works for me :+1: .  Does the location and size seem good?  The icon being so small it will be limited by pixels available.  I'll have to do pixel art for it but I can try my best to make it look like a mouse.

**Comment by samrocketman** [comment]
Here's my best attempt at pixel art.  We might need a real graphic artist to provide a sensible asset and I just sling the code for the feature.

![Screenshot taken at 2023-02-10_16-31-45](https://user-images.githubusercontent.com/875669/218202904-8f1b4db7-c413-46b8-bcfc-92f3102f802a.png)

**Comment by samrocketman** [comment]
Whatever the icon it should be constrained to the same dimensions as https://github.com/endless-sky/endless-sky/blob/master/images/ui/fast%20forward.png

**Comment by samrocketman** [comment]
Here's that pixel art zoomed in 
![Screenshot taken at 2023-02-10_16-35-35](https://user-images.githubusercontent.com/875669/218203429-2b981852-be85-4fe4-93c1-a883bedaf79c.png)

**Comment by Amazinite** [comment]
I don't think that adding an icon to the top left is the best place to denote that your mouse is now doing something it wasn't before. If anything, I think your mouse cursor itself should be what changes. Although since we don't have a custom mouse cursor when in game, it could be something that appears around the mouse cursor instead.

Mouse cursors changing to match the context of what would happen if you clicked is common OS behavior, and can also be seen in other games, so I think doing the same here would be easier to understand than adding a new icon to the corner of the screen.

**Comment by samrocketman** [comment]
Are those icons in base endless sky?  Or are those distro cursors?

**Comment by RisingLeaf** [comment]
> Are those icons in base endless sky? Or are those distro cursors?

I dont think they are ES

It should not be too difficult to change cursor with SDL, so I +1 that method

**Comment by Wyrdean** [comment]
"(have some Command and Conquer cursors and their contexts for reference)"
From the game command and conquer

**Comment by samrocketman** [comment]
I'm open to a target cursor or something like that.  I'm not familiar with that so would probably need some help on the coding side.

**Comment by RisingLeaf** [comment]
this would probably be the function to use: https://wiki.libsdl.org/SDL2/SDL_SetCursor

**Comment by Terin** [comment]
![](https://cdn.discordapp.com/attachments/935734693008138270/1073752603030196315/image.png)

Wyrdean on the discord put together a pretty decent icon for the original purpose, which I quite like. Replacing the mouse cursor would probably be ideal, but does seem like it could be tricky to get working across all platforms.

**Comment by Wyrdean** [comment]
If the idea to replace the cursor when mouse movement is enabled is confirmed, then I may as well make a set of cursor sprites for each of the mouse states. (Mouse aim, Standard, Highlighting a button etc) 
Current idea is to base it on MZ's original UI pieces such as the hailing/opening screen. Techno Wires, Cubes, Lines

**Comment by samrocketman** [comment]
We can probably restore the cursor with

```
SDL_Cursor *UI::DefaultCursor() {
	return SDL_GetDefaultCursor();
}
```
